storage/watchable_store.go

```diff
@@ -453,10 +453,10 @@ func unsafeAddWatcher(synced *map[string]map[*watcher]struct{}, k string, wa *wa
 	return nil
 }
 
-// newWatcherToEventMap creates a map that has watcher as key and events as
-// value. It enables quick events look up by watcher.
-func newWatcherToEventMap(sm map[string]map[*watcher]struct{}, evs []storagepb.Event) map[*watcher][]storagepb.Event {
-	watcherToEvents := make(map[*watcher][]storagepb.Event)
+// newWatcherToEventMap creates a map that has watcher as key and WatchResponse
+// as value. It enables quick events look up by watcher.
+func newWatcherToEventMap(sm map[string]map[*watcher]struct{}, evs []*storagepb.Event) map[*watcher]*storagepb.WatchResponse {
```

Comment:
I think this should be kept as `map[*watcher][]storagePb.Event` since it seems the map's `*watcher` key's `id` will always match the `WatchResponse.WatchId` (that is, there's enough information to make the WatchResponse structure at lookup time). Plus there would be less code churn overall.